Sources/Transaction/SignedContingentInputBuilderUtils.swift

```diff
@@ -0,0 +1,213 @@
+//
+//  Copyright (c) 2020-2022 MobileCoin. All rights reserved.
+//
+// swiftlint:disable multiline_function_chains
+
+import Foundation
+import LibMobileCoin
+
+enum SignedContingentInputBuilderUtils {
+
+    static func addRequiredOutput(
+        ptr: OpaquePointer,
+        publicAddress: PublicAddress,
+        amount: Amount,
+        rng: MobileCoinRng
+    ) -> Result<TxOut, TransactionBuilderError> {
+        var confirmationNumberData = Data32()
+
+        func ffiInner(
+            publicAddressPtr: UnsafePointer<McPublicAddress>,
+            rngCallbackPtr: UnsafeMutablePointer<McRngCallback>?
+        ) -> Result<Data, TransactionBuilderError> {
+            confirmationNumberData.asMcMutableBuffer { confirmationNumberPtr in
+                Data.make(withMcDataBytes: { errorPtr in
+                    mc_signed_contingent_input_builder_add_required_output(
+                        ptr,
+                        amount.value,
+                        amount.tokenId.value,
+                        publicAddressPtr,
+                        rngCallbackPtr,
+                        confirmationNumberPtr,
+                        &errorPtr)
+                }).mapError {
+                    switch $0.errorCode {
+                    case .invalidInput:
+                        return .invalidInput("\(redacting: $0.description)")
+                    case .attestationVerificationFailed:
+                        return .attestationVerificationFailed("\(redacting: $0.description)")
```

Comment:
commented on the other PR, but I don't think this function can throw an attestation error

Reply:
removed